.github/workflows/tests-emulated.yaml

```diff
@@ -0,0 +1,74 @@
+name: Run tests emulated
+
+on:
+  # Trigger the workflow on master but also allow it to run manually.
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - arm_action # TODO remove
+
+jobs:
+  run_tox_emulate:
+    name: tox -e py38_cython (${{ matrix.architecture }})
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        architecture:
+          - s390x
+          - arm64v8
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Cache PIP
+        uses: actions/cache@v2
+        with:
+          path: |
+            .pip
+          key: python-${{ matrix.architecture }}
+
+      - name: Set up emulation
+        run: |
+          docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+
+      - name: Run tox s390x
+        if: ${{ matrix.architecture == 's390x' }}
+        uses: docker://s390x/python:3.8-buster
+        env:
+          PIP_CACHE_DIR: /github/workspace/.pip/
+        with:
+          args: |
+            /bin/bash -c "
+            lscpu &&
+            mkdir -p $PIP_CACHE_DIR &&
+            chown -R $(whoami) $PIP_CACHE_DIR &&
+            pip install -U pip tox &&
+            python --version &&
+            pip --version &&
+            tox --version &&
+            tox -e py38_cython"
+
+      - name: Run tox arm64v8
+        if: ${{ matrix.architecture == 'arm64v8' }}
+        uses: docker://arm64v8/python:3.8-buster
```

Comment:
You could probably just template these on `matrix.architecture`, but skipping `if` and the copypasted blocks altogether?

Reply:
nope, I've tried and if fails if you template uses. both for actions and for docker. See this try [here](https://github.com/CaselIT/falcon/actions/runs/200257229) (scroll down)
That would be too easy I guess.

There is also no support for yaml anchors as of yet.
We could add an `sh` script, but I'm not sure where the best for it could be.

Reply:
Aha, nvm then, it works as-is.

Reply:
We may be able to use this new feature though https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action

Probably best to experiment on a follow on pr though

Reply:
But I guess you still cannot template `uses`, so it may be a moot point